### ISSUE TITLE
undefined is not an object (evaluating 'this.epoch=null')

### ISSUE THREAD

**Issue by johnfumaster** [spencermountain/spacetime#176]
**Error** 

`undefined is not an object (evaluating 'this.epoch=null')`

I'm receiving errors from our monitoring software that people are experiencing.  I don't know what's causing it.  I've thrown invalid dates, invalid strings, NaN into `spacetime()`, and I can't get it to throw this error.  

The line reports is coming from: webpack:///./node_modules/spacetime/builds/spacetime.js:3826:0:in 'epoch'

**Comment by spencermountain** [comment]
thanks john, that sounds weird.
Are you using it with .bind() or something?  it's a pretty [generic this call](https://github.com/spencermountain/spacetime/blob/master/builds/spacetime.js#L3826). `this` should always exist, unless being called in a fancy way.

**Comment by johnfumaster** [comment]
No, not any special way.  I am doing something like this mostly.
`spacetime(date).isValid()`

where `date` can be of `new Date()`, `new Date('2016-01-01')`, a number like `1488370835081`, `NaN`, `null`, `undefined` etc.

**Comment by johnfumaster** [comment]
I've just put in a try catch around the call to instantiate a spacetime to log the input that is causing the error.  I'll let you know my findings as soon as I find out.

**Comment by johnfumaster** [comment]
Ok. Here's what caused the errors. The code was instantiating with:
`let s = spacetime(”2019-11-21”)`
As well as 
`let a = spacetime("Thu Nov 21 2019 15:48:12 GMT+1100 (AEDT)")`

They seem to produce dates, it's just that spacetime code throws an error for some bizarre reason. 

Any ideas?

**Comment by johnfumaster** [comment]
Hi, we can finally replicate the error. It seems to be an issue with Safari 11 on desktop.

**Comment by spencermountain** [comment]
thanks John, that's really helpful.
I'm happy to fix this. The second date appears not parse into a valid date. 
If there's a weird browser issue, we should be able to set some kind of guard or polyfill.

what are the actual method calls, to reproduce it?

**Comment by spencermountain** [comment]
yeah, it seems safari has [some opinions](https://stackoverflow.com/questions/4310953/invalid-date-in-safari) in it's Date Object. That could be it.
Either way, the no-this error shouldn't happen. We should figure it out.
cheers

**Comment by johnfumaster** [comment]
Thanks Spencer. Much appreciated.  

The stack overflow only reports issues with 'yyyy-MM-dd' format.  But we are also getting errors when using `1574312228115` and `Mon Nov 25 2019 11:00:00 GMT+1100 (AEDT)`.
